tensorflow_graphics/projects/mesh_rcnn/ops/cubify_test.py

```diff
@@ -0,0 +1,249 @@
+            [0, 4, 1],
+            [0, 3, 4],
+            [1, 2, 10],
+            [2, 11, 10],
+            [1, 5, 2],
+            [1, 4, 5],
+            [2, 13, 11],
+            [2, 5, 13],
+            [3, 12, 14],
+            [3, 14, 6],
+            [3, 7, 4],
+            [3, 6, 7],
+            [14, 15, 7],
+            [14, 7, 6],
+            [4, 8, 5],
+            [4, 7, 8],
+            [15, 16, 8],
+            [15, 8, 7],
+            [5, 16, 13],
+            [5, 8, 16],
+            [9, 10, 17],
+            [10, 18, 17],
+            [17, 18, 20],
+            [18, 21, 20],
+            [9, 17, 20],
+            [9, 20, 12],
+            [10, 11, 18],
+            [11, 19, 18],
+            [18, 19, 21],
+            [19, 22, 21],
+            [11, 22, 19],
+            [11, 13, 22],
+            [20, 21, 23],
+            [21, 24, 23],
+            [12, 20, 23],
+            [12, 23, 14],
+            [23, 24, 15],
+            [23, 15, 14],
+            [21, 22, 24],
+            [22, 25, 24],
+            [24, 25, 16],
+            [24, 16, 15],
+            [13, 25, 22],
+            [13, 16, 25],
+        ], dtype=tf.float32
+    )
+
+    self.assertAllClose(expected_vertices_full, vertices[1])
+    self.assertAllClose(expected_faces_full, faces[1])
+
+  def test_cubify_on_sphere(self):
+    """Tests cubify with a voxelized sphere."""
+
+    test_data_dir = os.path.join(os.path.dirname(__file__), 'test_data')
+
+    shpere_voxel_path = os.path.join(test_data_dir, 'sphere_voxel.npy')
+    sphere = np.load(shpere_voxel_path).astype(np.float32)
+    sphere_tensor = tf.convert_to_tensor(np.expand_dims(sphere, 0))
+
+    sphere_vertices_path = os.path.join(test_data_dir, 'sphere_vertices.npy')
+    sphere_faces_path = os.path.join(test_data_dir, 'sphere_faces.npy')
+    expected_sphere_vertices = tf.convert_to_tensor(
+        np.load(sphere_vertices_path))
+    expected_sphere_faces = tf.convert_to_tensor(
+        np.load(sphere_faces_path))
+
+    mesh = cubify(sphere_tensor, 0.5)
+    vertices, faces = mesh.get_unpadded()
+    self.assertAllClose(expected_sphere_faces, faces[0])
+    self.assertAllClose(expected_sphere_vertices, vertices[0])
+
+  def test_no_batch_dimensions(self):
+    """Tests cubify on input voxel grid of shape `[2, 2, 2]` with random
+    values."""
+
+    voxel_grid = tf.random.uniform((2, 2, 2), 0, 1)
+
+    meshes = cubify(voxel_grid, 0.0)
+
+    self.assertEqual([1, 26, 3], meshes.get_padded()[0].shape)
+    self.assertEqual([1, 48, 3], meshes.get_padded()[1].shape)
+
+  def test_multiple_batch_dimensions(self):
+    """Tests cubify on input voxel grid of shape `[2, 3, 2, 2, 2]` with random
+    values."""
+
+    voxel_grid = tf.random.uniform((2, 3, 2, 2, 2), 0, 1)
+    meshes = cubify(voxel_grid, 0.0)
+
+    self.assertEqual([2, 3, 26, 3], meshes.get_padded()[0].shape)
+    self.assertEqual([2, 3, 48, 3], meshes.get_padded()[1].shape)
+
+  def test_raises_on_invalid_input(self):
```

Comment:
see above comment for example, these can also be combined using parameterised tests and helper methods ofTF-G